pkg/lib/oauth/redis/store.go

```diff
@@ -223,18 +298,24 @@ func (s *Store) ListOfflineGrants(userID string) ([]*oauth.OfflineGrant, error)
 		}
 
 		for id := range sessionList {
-			grant := &oauth.OfflineGrant{}
-			err = s.load(conn, offlineGrantKey(string(s.AppID), id), grant)
+			var data []byte
+			data, err = s.loadData(conn, offlineGrantKey(string(s.AppID), id))
 			if errors.Is(err, oauth.ErrGrantNotFound) {
 				_, err = conn.HDel(ctx, listKey, id).Result()
 				if err != nil {
 					s.Logger.WithError(err).Error("failed to update session list")
 				}
 			} else if err != nil {
 				return err
-			} else {
-				grants = append(grants, grant)
 			}
+
+			var grant *oauth.OfflineGrant
+			grant, err = s.unmarshalOfflineGrant(data)
+			if err != nil {
+				return err
+			}
+
+			grants = append(grants, grant)
```

Comment:
The original `grants = append(grants, grant)` logic is in the `else` block, we didn't return in the `ErrGrantNotFound` case. So we need to put back `unmarshalOfflineGrant` logic in the `else` block, or put `continue` in the `ErrGrantNotFound` case?